### ISSUE TITLE
Tor SOCKS proxy support for .onion hosts

### ISSUE THREAD

**Issue by teran-mckinney** [monero-ecosystem/monero-python#55]
Would be nice to have Tor SOCKS proxy support when using a .onion as a RPC host.

If you like, I can do a pull request for it. This is how I've done it in the past: https://github.com/sporestack/sporestack-python/blob/397c1f272c57fb08e369e0ddbfe5fd1172d041ad/sporestackv2/api_client.py#L56

Lots of ways to implement it. Even just to detect .onions and only try it for those.

I think #54 should be merged first and it would likely conflict.

What do you think?

Thank you!

**Comment by emesik** [comment]
Interesting. I realized I've never tried to connect from Pytohn code to an onion daemon, even though I run one myself.
Will check that, definitely.

**Comment by teran-mckinney** [comment]
Hey! Have you had a chance to take a look?

**Comment by emesik** [comment]
OK, I tried to start entire process in `torsocks` wrapper:

`$ torsocks python utils/daemoninfo.py xmrag4hf5xlabmob.onion:18081`

It worked.

Howewer, I guess what you want to achieve is to select what goes through Tor from the Python code itself? Perhaps the most flexible way would be to add an optional `proxies` keyword to the JSONRPC backends. A subclass could then implement all logic of selecting what to pass through Tor and what not.

**Comment by teran-mckinney** [comment]
Thanks for looking into this. Sure, I think that would be ideal. That way you could use Tor or any other SOCKS proxy.

**Comment by teran-mckinney** [comment]
Thank you! I appreciate it.

**Comment by teran-mckinney** [comment]
Seems like JSONRPCWallet doesn't support proxy_url but JSONRPCDaemon does. Is that intentional?

**Comment by emesik** [comment]
I just considered it unlikely someone would connect over proxy to their own wallet. But perhaps I'm missing something. It could be added easily in next release if there's benefit from such option.

**Comment by teran-mckinney** [comment]
Thanks for getting back to me.

That is exactly what I was going to use. I have the wallet RPC service listening on my Monero node and other servers accepting payments communicate with it over a Tor Hidden Service. I'm sure there's better designs though.

Right now I'm using socat, binding the hidden service to localhost and talking to that.

**Comment by emesik** [comment]
0.7.1 fixes this

**Comment by teran-mckinney** [comment]
Thank you!!
